File: objc/Facade/Facade/DataCompressor.m
#import "DataCompressor.h"

@implementation DataCompressor

-(NSString*)compress:(NSString*)data {
    switch (_compressionAlgo) {
        case CompressionAlgorithm_GZIP:
            return data;
        case CompressionAlgorithm_BZIP:
            return data;
        default:
            return nil;
    }
}

@end

File: swift/Facade/Facade/DataCompressor.swift
import Foundation

public class DataCompressor {
    private var compressorAlgo: CompressionAlgorithm
    
    init(_ algo: CompressionAlgorithm) {
        self.compressorAlgo = algo
    }
    
    func compress(_ data: String) -> String? {
        switch compressorAlgo {
        case .CompressionAlgorithm_GZIP:
            return data
        case .CompressionAlgorithm_BZIP:
            return data
        default:
            return nil
        }
    }
}
